pennylane/workflow/execution.py

```diff
@@ -583,6 +583,8 @@ def cost_fn(params, x):
         if config.mcm_config.postselect_mode == "hw-like":
             raise ValueError("Using postselect_mode='hw-like' is not supported with jax-jit.")
         config.mcm_config.postselect_mode = "fill-shots"
+    if config.mcm_config.mcm_method == "single-branch-statistics" and not qml.compiler.active():

```

Comment:
I'm wondering if I even need to check if there is an active compiler or not here, since `qml.execute` shouldn't be getting hit when qjit-ing. Also would like thoughts on if it makes more sense for this check to be in `QNode.__call__` rather than `qml.execute`.